servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -640,12 +663,12 @@ previous, new ConnState(newList, newState))) {
             LOGGER.trace("Load balancer for {}: added a new connection {} to {} after {} attempt(s).",
                     targetResource, connection, this, addAttempt);
             // Instrument the new connection so we prune it on close
-            connection.onClosing().beforeFinally(() -> {
+            connection.onClose().beforeFinally(() -> {
```

Comment:
while this is preferred to remove connections from the selection process we still need to manage the lifecycle of the connections. we can use `connection.onClosing` in the future to disable selection eligibility but the connection by adding extra state or another collection but this PR prioritizes correctness and we can followup to optimize.